### PR TITLE
Add dhclient basic configuration

### PR DESCRIPTION
**Pull request by avlasov-mos-de**
Allows configuring general section as well as configuring
    each dhcp enabled interface separetly.
    Does not allow alias or lease configuration.

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

File: linux/network/dhclient.sls [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{%- from "linux/map.jinja" import network with context %}
+{%- if network.enabled %}
+
+{%- set dhclient = network.get('dhclient_config', False) %}
```
Comment (fpytloun) [inline linux/network/dhclient.sls]: Name the key only `dhclient` to match your newly introduced role

File: linux/map.jinja [hunk 2]
```diff
@@ -110,6 +110,7 @@
            'host': 'none',
         },
         'host': {},
+        'dhclient_cfg_path': '/etc/dhcp/dhclient.conf',
```
Comment (fpytloun) [inline linux/map.jinja]: I think dhclient_config would be sufficient but not mandatory.

File: README.rst [hunk 3]
```diff
@@ -772,6 +772,73 @@ OpenVswitch Bridges
             use_interfaces:
             - eth1
 
+DHCP client configuration
```
Comment (fpytloun) [inline README.rst]: Also add new tests into `test/pillar` and check `make test`

File: linux/files/dhclient.conf [hunk 4]
```diff
@@ -0,0 +1,91 @@
+{# Macro definition #}
+{%- macro render_key(section, key) -%}
+{%- if section[key] is defined and section[key]|length > 0 %}
```
Comment (fpytloun) [inline linux/files/dhclient.conf]: don't know if `section[key] is defined` is working as you expect, I would instead use `section.get(key)` that defaults to None.

**Comment by avlasov-mos-de** [comment]
Thanks for feedback, I did  a couple of changes accordingly to comments. Also I rewrote macro a bit in order to support nested lists for some options. As for tests, I updated pillar/network.sls as it looks like this change has to be tested as part of network. make test finished with "Execution successful".  If I got it wrong could you explain what kind of tests I should add

**Pull request merged**